File: Bai_96/main.cpp
#include<bits/stdc++.h>
using namespace std;

bool valid_number(const string &number, const set<char> &digits) {
    for(char digit : number)
        if(digits.find(digit) != digits.end())
            return false;
    return true;
}

int find_number(int P, const set<char> &digits){
    for(int num = P; num > 0; --num){
        string str_num = to_string(num);
        if (valid_number(str_num, digits))
            return num;
    }
    return -1;
}

int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    freopen("SODB.INP","r",stdin);
    freopen("SODB.OUT","w",stdout);
    int P; cin >> P;
    string S; cin >> S;
    set<char> digits(S.begin(), S.end());
    int result = find_number(P, digits);
    cout << result << endl;
    fclose(stdin);
    fclose(stdout);
    return 0;
}

File: Bai_96/main.py
def valid_number(number, S):
    for digit in S:
        if digit in number:
            return False
    return True

def find_number(P, S):
    str_P = str(P)
    length_P = len(str_P)
    for num in range(P, 0, -1):
        str_num = str(num)
        if valid_number(str_num, S):
            return num
    return -1

with open('SODB.INP', 'r') as file:
    P = int(file.readline().strip())
    S = file.readline().strip()
with open('SODB.OUT', 'w') as file:
    result = find_number(P, S)
    file.write(str(result) + '\n')